db/version_edit_handler.cc

```diff
@@ -892,42 +907,73 @@ Status ManifestTailer::VerifyFile(const std::string& fpath,
 }
 
 void DumpManifestHandler::CheckIterationResult(const log::Reader& reader,
-                                               Status* s) {
-  VersionEditHandler::CheckIterationResult(reader, s);
+                                               Status* s,
+                                               std::vector<FileMetaData*>*,
+                                               uint64_t sst_file_number) {
+  std::vector<FileMetaData*>* files = new std::vector<FileMetaData*>;
+  bool found = false;
+  if (sst_file_number != 0) {
+    VersionEditHandler::CheckIterationResult(reader, s, files);
```

Comment:
Still trying to understand more about this PR, but at the moment, I am not fully convinced of the necessity of a separate `CheckIterationResult`.